server/app/repository/ApplicationRepository.java

```diff
@@ -128,14 +129,16 @@ private Application submitApplicationInternal(
           continue;
```

Comment:
Do we need to loop through old applications if we only care about comparing with the most recent one? Or is it possible that there could be multiple `ACTIVE` applications?

Reply:
That's a good question. Based on the code here, it looks like we mark all the old apps as `OBSOLETE` before saving the new one as `ACTIVE` so there should only ever be one `ACTIVE` version. This loop ignores `OBSOLETE` so those would never match and the `DRAFT` application is the one we're looking at, so that would be skipped as well. The only other lifecycle stage is `DELETED` which is never used. So, long story short, I think we're only ever comparing to the single `ACTIVE` application in the db in this loop. cc @yotommy who originally added the dupes check here. What do you think?